test/e2e/deployment.go

```diff
@@ -763,6 +764,75 @@ func testRollbackDeployment(f *framework.Framework) {
 	Expect(err).NotTo(HaveOccurred())
 }
 
+// testDryRunRollbackDeployment tests that a deployment is created (revision 1) and updated (revision 2), and
+// then rollback with dry-run option (should just output template of revision 1).
+func testDryRunRollbackDeployment(f *framework.Framework) {
```

Comment:
Sorry, all I meant is you should add a call to [test-cmd.sh](https://github.com/kubernetes/kubernetes/blob/2352464853faac92a3c9379f7e546d3c06938984/hack/make-rules/test-cmd.sh#L1958) and not a whole e2e test.

Something like:
```sh
 # Rollback to revision 1 - should be no-op
  kubectl rollout undo deployment nginx --to-revision=1 "${kube_flags[@]}"
  kube::test::get_object_assert deployment "{{range.items}}{{$deployment_image_field}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
  # Update the deployment (revision 2)
  kubectl apply -f hack/testdata/deployment-revision2.yaml "${kube_flags[@]}"
  kube::test::get_object_assert deployment.extensions "{{range.items}}{{$deployment_image_field}}:{{end}}" "${IMAGE_DEPLOYMENT_R2}:"

# YOUR ADDITIONS
kubectl rollout undo deploy/nginx --dry-run | grep nginx:test-cmd
# verify we didn't actually rollback
kubectl get deploy/nginx -o yaml | grep nginx:1.7.9


# Rollback to revision 1